# Most Popular Amenities In NC area

In [340]:
import pandas as pd
import numpy as np
import os
import glob
import altair as alt
import plotly.graph_objects as go
import plotly.express as px
import json

In [341]:
df1=pd.read_csv("../data/processed/clean_airbnb.gz")
#df1=df1[df['host_location'].str.split(', ').str[1] == 'NC']
df1.head()

id                                               name  price  \
0  108061  Walk to stores/parks/downtown. Fenced yard/Pet...  100.0   
1  108061  Walk to stores/parks/downtown. Fenced yard/Pet...  100.0   
2  108061  Walk to stores/parks/downtown. Fenced yard/Pet...  100.0   
3  108061  Walk to stores/parks/downtown. Fenced yard/Pet...  100.0   
4  108061  Walk to stores/parks/downtown. Fenced yard/Pet...  100.0   

         date available        room_type  accommodates  bedrooms  beds  \
0  2023-03-19         f  Entire home/apt             2       1.0   1.0   
1  2023-03-20         f  Entire home/apt             2       1.0   1.0   
2  2023-03-21         f  Entire home/apt             2       1.0   1.0   
3  2023-03-22         f  Entire home/apt             2       1.0   1.0   
4  2023-03-23         f  Entire home/apt             2       1.0   1.0   

                                           amenities  ...  \
0  ["Coffee maker", "Cleaning products", "Extra p...  ...   
1  ["Coffee maker", "Cleaning products", "Extra p...  ...   
2  ["Coffee maker", "Cleaning products", "Extra p...  ...   
3  ["Coffee maker", "Cleaning products", "Extra p...  ...   
4  ["Coffee maker", "Cleaning products", "Extra p...  ...   

   host_acceptance_rate  minimum_nights maximum_nights  number_of_reviews  \
0                   71%              30            365                 92   
1                   71%              30            365                 92   
2                   71%              30            365                 92   
3                   71%              30            365                 92   
4                   71%              30            365                 92   

  review_scores_rating reviews_per_month calculated_host_listings_count  \
0                 4.51              0.66                              2   
1                 4.51              0.66                              2   
2                 4.51              0.66                              2   
3                 4.51              0.66                              2   
4                 4.51              0.66                              2   

  calculated_host_listings_count_entire_homes  \
0                                           2   
1                                           2   
2                                           2   
3                                           2   
4                                           2   

  calculated_host_listings_count_private_rooms  \
0                                            0   
1                                            0   
2                                            0   
3                                            0   
4                                            0   

   calculated_host_listings_count_shared_rooms  
0                                            0  
1                                            0  
2                                            0  
3                                            0  
4                                            0  

[5 rows x 28 columns]

In [342]:
df2=pd.read_csv("../data/processed/clean_listings.gz")
#df=df[df['host_location'].str.split(', ').str[1] == 'NC']
df2.head()

id                                               name  host_id  \
0  108061  Walk to stores/parks/downtown. Fenced yard/Pet...   320564   
1  155305                 Cottage! BonPaul + Sharky's Hostel   746673   
2  156805                  Private Room "Ader" at BPS Hostel   746673   
3  197263                       Tranquil Room & Private Bath   961396   
4  209068                                    Terrace Cottage  1029919   

   host_since  host_response_time host_response_rate host_acceptance_rate  \
0  2010-12-16      within an hour               100%                  71%   
1  2011-06-26      within an hour               100%                  99%   
2  2011-06-26      within an hour               100%                  99%   
3  2011-08-12  within a few hours               100%                  93%   
4  2011-08-28  within a few hours               100%                  44%   

   host_total_listings_count  latitude  longitude  ... availability_365  \
0                        3.0  35.60670  -82.55563  ...              246   
1                       11.0  35.57864  -82.59578  ...              197   
2                       11.0  35.57864  -82.59578  ...                0   
3                        2.0  35.57808  -82.63689  ...               88   
4                        1.0  35.61856  -82.55276  ...              231   

  number_of_reviews  review_scores_rating  review_scores_accuracy  \
0                92                  4.51                    4.56   
1               383                  4.59                    4.72   
2                67                  4.52                    4.73   
3                66                  4.95                    4.97   
4                60                  4.87                    4.84   

   review_scores_cleanliness  review_scores_checkin  \
0                       4.70                   4.85   
1                       4.42                   4.83   
2                       4.43                   4.76   
3                       5.00                   4.95   
4                       4.84                   5.00   

   review_scores_communication  review_scores_location       city  \
0                         4.80                    4.84  Asheville   
1                         4.74                    4.93  Asheville   
2                         4.61                    4.84  Asheville   
3                         4.92                    4.82  Asheville   
4                         4.98                    4.93  Asheville   

            state  
0  North Carolina  
1  North Carolina  
2  North Carolina  
3  North Carolina  
4  North Carolina  

[5 rows x 31 columns]

In [343]:
df1_unique = df1.drop_duplicates(subset='id', keep='first')

In [344]:
amenities =df1_unique['amenities']
# Create an empty dictionary to store the word counts
amenities_counts = {}

# Loop through each sublist in the list of lists
for sublist in amenities:
    # Loop through each word in the sublist
    for word in json.loads(sublist):
        # If the word is already in the dictionary, increment its count
        if word in amenities_counts:
            amenities_counts[word] += 1
        # Otherwise, add the word to the dictionary with a count of 1
        else:
            amenities_counts[word] = 1

# Print the word counts
print(amenities_counts)

{'Coffee maker': 1544, 'Cleaning products': 1247, 'Extra pillows and blankets': 1768, 'Pets allowed': 905, 'Cooking basics': 1976, 'Indoor fireplace: gas': 162, 'Dishes and silverware': 2456, 'Private entrance': 1982, 'Free street parking': 1023, 'Essentials': 2653, 'Single level home': 729, 'Free parking on premises': 2238, 'Freezer': 1339, 'Dining table': 1218, 'Refrigerator': 2199, 'Free dryer – In unit': 701, 'Induction stove': 18, 'Wifi': 2377, 'Carbon monoxide alarm': 2366, 'Blender': 592, 'Hangers': 2462, 'Central air conditioning': 972, 'Hot water kettle': 974, 'Microwave': 2340, 'Books and reading material': 773, 'Patio or balcony': 520, 'Long term stays allowed': 1163, 'Central heating': 702, 'Hot water': 2432, 'Room-darkening shades': 880, 'Toaster': 1237, 'Kitchen': 2252, 'Wine glasses': 1287, 'Ceiling fan': 1178, 'Host greets you': 86, 'Smoke alarm': 2701, 'Baking sheet': 1013, 'BBQ grill': 737, 'Courtyard view': 39, 'Oven': 1406, 'Dishwasher': 1504, 'Clothing storage': 21

In [345]:
# Sort the dictionary based on its values in descending order
sorted_dict = dict(sorted(amenities_counts.items(), key=lambda item: item[1], reverse=True))

amenity = pd.DataFrame(list(sorted_dict.items()), columns=['name', 'count'])


top10=amenity.loc[:10]

In [449]:
domains = ['Smoke alarm', 'Hangers', 'Dishes and silverware', 'Microwave', 'Wifi']
color_scale = alt.Scale(
    domain=domains,
)


shape_scale = alt.Scale(
    domain=domains,
    range=[
    #alarm
    "M8.5 5.5a.5.5 0 0 0-1 0v3.362l-1.429 2.38a.5.5 0 1 0 .858.515l1.5-2.5A.5.5 0 0 0 8.5 9V5.5z M6.5 0a.5.5 0 0 0 0 1H7v1.07a7.001 7.001 0 0 0-3.273 12.474l-.602.602a.5.5 0 0 0 .707.708l.746-.746A6.97 6.97 0 0 0 8 16a6.97 6.97 0 0 0 3.422-.892l.746.746a.5.5 0 0 0 .707-.708l-.601-.602A7.001 7.001 0 0 0 9 2.07V1h.5a.5.5 0 0 0 0-1h-3zm1.038 3.018a6.093 6.093 0 0 1 .924 0 6 6 0 1 1-.924 0zM0 3.5c0 .753.333 1.429.86 1.887A8.035 8.035 0 0 1 4.387 1.86 2.5 2.5 0 0 0 0 3.5zM13.5 1c-.753 0-1.429.333-1.887.86a8.035 8.035 0 0 1 3.527 3.527A2.5 2.5 0 0 0 13.5 1z",
    #hanger
    "M13 10.551v-.678A4.005 4.005 0 0 0 16 6c0-2.206-1.794-4-4-4S8 3.794 8 6h2c0-1.103.897-2 2-2s2 .897 2 2-.897 2-2 2a1 1 0 0 0-1 1v1.551l-8.665 7.702A1.001 1.001 0 0 0 3 20h18a1.001 1.001 0 0 0 .664-1.748L13 10.551zM5.63 18 12 12.338 18.37 18H5.63z",
    #dish washer
    "M20.9999 3.99999L21 20C21 21.1046 20.1046 22 19 22H5C3.89543 22 3 21.1046 3 20V4C3 2.89543 3.89543 2 5 2H18.9999C20.1045 2 20.9999 2.89543 20.9999 3.99999Z M18 5.01L18.01 4.99889 M12 19C15.3137 19 18 16.3137 18 13C18 9.68629 15.3137 7 12 7C8.68629 7 6 9.68629 6 13C6 16.3137 8.68629 19 12 19Z M12 16C10.3431 16 9 14.6569 9 13 M18 5.01L18.01 4.99889",
    #mircrowave
    "M6.8,10.61L5.37,9.19C5.73,8.79,6.59,8,7.75,8c0.8,0,1.39,0.39,1.81,0.67C9.87,8.88,10.07,9,10.25,9 c0.37,0,0.8-0.41,0.95-0.61l1.42,1.42c-0.36,0.4-1.22,1.19-2.37,1.19c-0.79,0-1.37-0.38-1.79-0.66C8.13,10.12,7.94,10,7.75,10 C7.38,10,6.95,10.41,6.8,10.61z M7.75,15c0.19,0,0.38,0.12,0.71,0.34c0.42,0.28,1,0.66,1.79,0.66c1.16,0,2.01-0.79,2.37-1.19 l-1.42-1.42c-0.15,0.2-0.59,0.61-0.95,0.61c-0.18,0-0.38-0.12-0.69-0.33C9.14,13.39,8.55,13,7.75,13c-1.16,0-2.02,0.79-2.38,1.19 l1.42,1.42C6.95,15.41,7.38,15,7.75,15z M22,6v12c0,1.1-0.9,2-2,2H4c-1.1,0-2-0.9-2-2V6c0-1.1,0.9-2,2-2h16C21.1,4,22,4.9,22,6z M14,6H4v12h10V6z M19,16c0-0.55-0.45-1-1-1c-0.55,0-1,0.45-1,1c0,0.55,0.45,1,1,1C18.55,17,19,16.55,19,16z M19,12 c0-0.55-0.45-1-1-1c-0.55,0-1,0.45-1,1c0,0.55,0.45,1,1,1C18.55,13,19,12.55,19,12z M19,7h-2v2h2V7z",
    #Wifi
    "M15.384 6.115a.485.485 0 0 0-.047-.736A12.444 12.444 0 0 0 8 3C5.259 3 2.723 3.882.663 5.379a.485.485 0 0 0-.048.736.518.518 0 0 0 .668.05A11.448 11.448 0 0 1 8 4c2.507 0 4.827.802 6.716 2.164.205.148.49.13.668-.049z M13.229 8.271a.482.482 0 0 0-.063-.745A9.455 9.455 0 0 0 8 6c-1.905 0-3.68.56-5.166 1.526a.48.48 0 0 0-.063.745.525.525 0 0 0 .652.065A8.46 8.46 0 0 1 8 7a8.46 8.46 0 0 1 4.576 1.336c.206.132.48.108.653-.065zm-2.183 2.183c.226-.226.185-.605-.1-.75A6.473 6.473 0 0 0 8 9c-1.06 0-2.062.254-2.946.704-.285.145-.326.524-.1.75l.015.015c.16.16.407.19.611.09A5.478 5.478 0 0 1 8 10c.868 0 1.69.201 2.42.56.203.1.45.07.61-.091l.016-.015zM9.06 12.44c.196-.196.198-.52-.04-.66A1.99 1.99 0 0 0 8 11.5a1.99 1.99 0 0 0-1.02.28c-.238.14-.236.464-.04.66l.706.706a.5.5 0 0 0 .707 0l.707-.707z"

    #"M17.671 14.307C16.184 12.819 14.17 12 12 12s-4.184.819-5.671 2.307l1.414 1.414c1.11-1.11 2.621-1.722 4.257-1.722 1.636.001 3.147.612 4.257 1.722l1.414-1.414z M20.437 11.292c-4.572-4.573-12.301-4.573-16.873 0l1.414 1.414c3.807-3.807 10.238-3.807 14.045 0l1.414-1.414z",
    ]
)

In [347]:
top5=top10.iloc[[0,2,3,6,8]]
test=top5.copy()
#top5['count'] = top5['count'].apply(np.log)

In [348]:
test['count']=round(top5['count'])/50

In [349]:
df_unmelted = test.loc[test.index.repeat(test['count'])].reset_index(drop=True)
df_unmelted

name  count
0    Smoke alarm  54.02
1    Smoke alarm  54.02
2    Smoke alarm  54.02
3    Smoke alarm  54.02
4    Smoke alarm  54.02
..           ...    ...
240    Microwave  46.80
241    Microwave  46.80
242    Microwave  46.80
243    Microwave  46.80
244    Microwave  46.80

[245 rows x 2 columns]

In [528]:
bars = alt.Chart(df_unmelted).mark_point(filled=True, opacity=1, size=8).encode(
    alt.X('x:O', axis=None),
    alt.Y('name:N'),
    alt.Shape('name:N', scale=shape_scale,title="Amenity Type",legend=alt.Legend(symbolSize=5,labelOverlap=False)),
    #alt.ShapeValue(shape_scale)
    alt.Color('name:N', scale=color_scale,title="Amenity Type"),
    #alt.legend(orient="left")
    #color=alt.value("#116EA1"),
).properties(width=1250, height=440).transform_window(
    x='rank()',
    groupby=['name'])
#.configure_axis(
#    labelFontSize=30,
#    titleFontSize=20,
#    grid=False
#)


#bars.configure_title(fontSize=14).configure(background='#D9E9F0')


bars.title ="Top 5 Counts Amenities"
bars.encoding.x.title = 'Count'
bars.encoding.y.title = 'Amenities'

#bars.configure_title(fontSize=14).configure(background='white')

bars

alt.Chart(...)

In [529]:
bars2 = alt.Chart(top5).mark_point(filled=True, opacity=1, size=20).encode(
    alt.X('count:Q',title="Count",scale=alt.Scale(domain=[1000,4500])),
    alt.Y('name:N',title="Amenities"),
    alt.Shape('name:N', scale=shape_scale),
    #alt.ShapeValue(shape_scale)
    alt.Color('name:N' ),
    #alt.legend(orient="left")
    #color=alt.value("#116EA1"),
).properties(width=950, height=340)


#bars.configure_title(fontSize=14).configure(background='#D9E9F0')


bars2.title ="Top 5 Counts Amenities"
#bars.encoding.x.title = 'Count'
#bars.encoding.y.title = 'Amenities'

#bars.configure_title(fontSize=30).configure(background='white')
bars2

alt.Chart(...)

In [354]:
domains = ['Smoke alarm', 'Hangers', 'Dishes and silverware', 'Microwave', 'Wifi']

price_amenty = df1_unique[['amenities','price']].reset_index(drop=True)

price_amenty = price_amenty[price_amenty['amenities'].apply(lambda x: any(item in x for item in domains))]

In [433]:
review_amenty = df1_unique[['amenities','review_scores_rating']].reset_index(drop=True)

In [439]:
price_amenty['tag'] = price_amenty['amenities'].apply(lambda x: 'Smoke alarm' if 'Smoke alarm' in x else '')
review_amenty['tag'] = review_amenty['amenities'].apply(lambda x: 'Smoke alarm' if 'Smoke alarm' in x else '')
def split_amenities(df):
    new_df=df.copy()
    c=1
    for i,row in df.iterrows():
        if i%100 == 0:
            #print(100*c)
            c+=1
        try:
            new_row =pd.DataFrame([df.iloc[i]]).copy()
            if 'Hangers' in row[0]:
                #print(1)
                new_row['tag'] = 'Hangers'
                new_df=pd.concat([new_df, new_row], axis=0)
            if 'Dishes and silverware' in row[0]:
                new_row['tag'] = 'Dishes and silverware'
                new_df=pd.concat([new_df, new_row], axis=0)
            if 'Microwave' in row[0]:
                #new_row = row.copy()
                new_row['tag'] = 'Microwave'
                new_df=pd.concat([new_df, new_row], axis=0)
            if 'Wifi' in row[0]:
                #new_row = row.copy()
                new_row['tag'] = 'Wifi'
                new_df=pd.concat([new_df, new_row], axis=0)
        except IndexError:
            break
    new_df=new_df[new_df['tag']!=""]
    #new_df = new_df[new_df['price']<3000]
    return new_df


z=split_amenities(price_amenty)
r=split_amenities(review_amenty)

In [389]:
alt.data_transformers.enable('json')

DataTransformerRegistry.enable('json')

In [522]:
boxs=alt.Chart(z).mark_boxplot(clip=True).encode(
    alt.X('tag:N',title="Amenities"),
    alt.Y('price:Q', title="Price",scale=alt.Scale(domain=[0,400])),
    alt.Color("tag:N",title="Amenity Type"),
).properties(width=1250, height=340)

boxs.title ="Boxplot of Price for Top 5 Counts Amenities"


In [521]:
boxs2=alt.Chart(r).mark_boxplot(clip=True).encode(
    alt.X('tag:N',title="Amenities"),
    alt.Y('review_scores_rating:Q', title="Review Scores Rating",scale=alt.Scale(domain=[3,5])),
    alt.Color("tag:N",title="Amenity Type"),
).properties(width=1250, height=340)

boxs2.title ="Boxplot of review scores rating for Top 5 Counts Amenities"


In [533]:
Chart=(boxs & bars & boxs2)
Chart

alt.VConcatChart(...)

Based on the graph, it is evident that the top 5 amenities that are most popular among Airbnb guests are Smoke alarm, Hangers, Dishes and silverware, Microwave, and Wifi. These amenities are all basic necessities that guests expect to find in a rental, and their availability can significantly impact the overall guest experience.

Furthermore, the graph shows that the Airbnb listings with these top 5 amenities have a similar price of around $200 and high review scores ratings around 4.85. This suggests that these amenities are essential for maintaining a competitive price point and ensuring guest satisfaction.

In conclusion, as an Airbnb host, it is crucial to offer these popular amenities to increase your rental's price and review scores. Guests are more likely to choose a rental with these amenities, which can result in more bookings and positive reviews. Therefore, investing in these amenities can be a smart move for hosts looking to improve their Airbnb business.